frontend/src/routes/Applications/CreateApplication/Subscription/controlData/utils.js

```diff
@@ -609,6 +609,7 @@ export const setAvailableSecrets = (control, result) => {
         }
     } else {
         control.isLoading = loading
+        if (!loading) control.isLoaded = true
```

Comment:
Control.isLoading is being updated before and after asynchronous loading is being performed. When `setAvailableSecrets` is first called, loading is in progress. Additional calls are made, but when there are no git secrets in the cluster, control.isLoaded is never updated.

 The idea here is that when control.isLoading is set to false (indicating that a fetch has finished), control.isLoaded needs to be updated to prevent subsequent calls, regardless of the contents of the results.